### ISSUE TITLE
Add "Cancel all" button for stalled transactions

### ISSUE THREAD

**Issue by danfinlay** [MetaMask/metamask-extension#4022]
Sometimes a user keeps submitting new transactions when the network is slow, then finds out these are all valid transactions!

A "cancel all" button would submit null transactions for all pending transactions, but with higher gas prices.

**Comment by cjeria** [comment]
We'd probably need to display a "total cost" for submitting null transactions but with higher gas price. This button might also be placed above history somewhere or at least within settings as a starting point while we figure out where to place this button in the new retry tx designs.

**Comment by CruzMolina** [comment]
Related to #3756. Can also occur when a very low gas price (1 Gwei) tx is attempted. Importing the wallet's mnemonic into another browser on the same system or the same browser on a different system clears up the clog and sends out all the valid stuck pending tx's.

**Comment by schmijos** [comment]
This would also be helpful for me to develop Dapps. I made a mistake and I've now 300 pending transactions 😢

**Comment by bdresser** [comment]
will probably be handled in https://github.com/MetaMask/Design/issues/25

**Comment by bdresser** [comment]
This should probably have a custom confirmation screen
* may want to show each tx hash that's getting cancelled (and links to etherscan) - this could replace the primary value up top? 
* we should suggest a gas high enough to beat out everything pending, but still allow the gas "edit" link
* what should we show as the recipient address? 
* "gas fee" and "total" will be the same, maybe we can remove one? ("total gas fee") ?
* should add an article somewhere about why cancelling requires gas
* could hard-code "CANCEL ALL" or "CANCEL PENDING TRANSACTIONS" in the pill that usually displays the function name

**Comment by bdresser** [comment]
from an older duplicate issue

![image](https://user-images.githubusercontent.com/1016190/45233618-2d4b3500-b288-11e8-8d32-56c665998175.png)

also, copying this wireframe from https://github.com/MetaMask/Design/issues/30 that I think could be very similar to what we could show for a "cancel all" feature. We could show the pending tx that'd be cancelled, show the gas it will require for each, then sum it up at the bottom. Probably don't want to allow de-selection as it'd cause some nonce issues. This would be a slightly more robust view than repurposing the Confirm screen, but it'd also give the user much more detail.

![image](https://user-images.githubusercontent.com/1016190/45234353-510f7a80-b28a-11e8-9133-6ea030cf6394.png)

So to sum up all this word-sauce, I think there are two options:
(1) repurpose the Confirm screen, try to cram in relevant info
(2) build a new screen that displays multiple tx; could be re-used for submitting batch transactions

**Comment by danfinlay** [comment]
re: step 4: We only need to increase the gas price >=10% over the replaced tx for all known node types to accept it as a replacement. Anything more is over-paying.

**Comment by danfinlay** [comment]
You know what would be great? If a single tx could span nonces. That way, a single "cancellation tx" with high gas could invalidate all of these other pending transactions.
